content/docs/5.1/concepts/explore-tikv-features/fault-tolerance.md

```diff
@@ -1,10 +1,201 @@
+    ```txt
+    Run finished, takes 11.722575701s
+    INSERT - Takes(s): 11.7, Count: 10000, OPS: 855.2, Avg(us): 18690, Min(us): 11262, Max(us): 61304, 99th(us): 36000, 99.9th(us): 58000, 99.99th(us): 62000
+
+    ```
+## Step 3. Verify the data importing.
+
+Within python3.5+ REPL environment, you can scan all the keys that just inserted by `go-ycsb` and assert the count of them is `recordcount` in the `go-ycsb` command above.
+```python
+>>> from tikv_client import RawClient
+>>> client = RawClient.connect("127.0.0.1:2379")
+>>> len(client.scan_keys(None, None, 10000))
+10000
+```
+
+The evaluation of the last expression should be **10000**, as the `recordcount` specified in the `go-ycsb` command.
+
+
+## Step 4. Run a sample workload.
+
+Go to the source directory of `go-ycsb`, use the following command to run the `workloada` from YCSB benchmark, simulating multiple client connections, each performing mixed 50% read and 50% write operations.
+
+```sh
+./bin/go-ycsb run tikv -P workloads/workloada -p tikv.pd="127.0.0.1:2379" -p tikv.type="raw" -p tikv.conncount=16 -p threadcount=16 -p recordcount=10000 -p operationcount=1000000
+```
+
+you'll see per-operation statistics print to standard output every second.
+
+```txt
+...
+READ   - Takes(s): 10.0, Count: 7948, OPS: 796.2, Avg(us): 395, Min(us): 72, Max(us): 20545, 99th(us): 2000, 99.9th(us): 19000, 99.99th(us): 21000
+UPDATE - Takes(s): 10.0, Count: 7945, OPS: 796.8, Avg(us): 19690, Min(us): 11589, Max(us): 40429, 99th(us): 34000, 99.9th(us): 41000, 99.99th(us): 41000
+READ   - Takes(s): 20.0, Count: 15858, OPS: 793.6, Avg(us): 380, Min(us): 68, Max(us): 20545, 99th(us): 2000, 99.9th(us): 3000, 99.99th(us): 21000
+UPDATE - Takes(s): 20.0, Count: 15799, OPS: 791.1, Avg(us): 19834, Min(us): 10505, Max(us): 41090, 99th(us): 35000, 99.9th(us): 40000, 99.99th(us): 41000
+...
+```
+
+Normally, this workload above will run several minutes, you will have enough time to check and manipulate the cluster.
+
+## Step 5. Check the workload
+
+1. Go to the [Grafana](https://grafana.com) page at [http://127.0.0.1:3000](http://127.0.0.1:3000)
+
+2. Login with default username `admin` and password `admin`
+
+3. Go to dashboard **playground-tikv-summary**, the OPS information is in panel **gRPC message count** in row **gRPC**.
+
+{{< figure
+    src="/img/docs/check-ops.png"
+    width="80"
+    number="1" >}}
+
+4. By default, TiKV replicates all data 3 times and balances it across all stores. To see this balance, go to page **playground-overview** and check the region count across all nodes. In this example, we load a small amount of data, thus only one region is presented:
+
+{{< figure
+    src="/img/docs/fault-tolerance-region-count.png"
+    width="80"
+    number="1" >}}
+
+## Step 6. Simulate a single node failure
+
+To understand fault tolerance in TiKV, it's important to review a few concepts from the [architecture](https://github.com/tikv/tikv#tikv-software-stack).
```

Comment:
links to https://tikv.org/docs/5.1/reference/architecture/overview/

Reply:
Noted.